src/OrchardCore.Modules/OrchardCore.ReCaptcha/Users/Handlers/RegistrationFormEventHandler.cs

```diff
@@ -23,9 +23,13 @@ public Task RegisteredAsync(IUser user)
 
     public async Task RegistrationValidationAsync(Action<string, string> reportError)
     {
-        if (await _signInManager.GetExternalLoginInfoAsync() == null)
+        // When logging in via an external provider, authentication security is already handled by the provider.
+        // Therefore, using a CAPTCHA is unnecessary and impractical, as users wouldn't be able to complete it anyway.
+        if (await _signInManager.GetExternalLoginInfoAsync() != null)
```

Comment:
I am wondering why this is missing the _reCaptchaService.IsThisARobot() check

Reply:
I don't know, but since that caused problems in the other case too (hence you removed it in `main`), I wouldn't add it.